lib/chore/job.rb

```diff
@@ -108,11 +115,22 @@ def perform(*args)
 
     # Use the current configured publisher to send this job into a queue.
     def perform_async(*args)
-      self.class.run_hooks_for(:before_publish,*args)
-      @chore_publisher ||= self.class.options[:publisher]
-      @chore_publisher.publish(self.class.prefixed_queue_name,self.class.job_hash(args))
-      self.class.run_hooks_for(:after_publish,*args)
+      push_to_publisher(args)
+    end
+
+    # Use the current configured publisher to send this job into a queue with the given delay period (in seconds). If
+    # the given publisher does not support delayed jobs a warning should be issued and the job will be queued without a
+    # delay.
+    def perform_delayed(delay, *args)
+      push_to_publisher(args, delay: delay)
     end
 
+    private
+    def push_to_publisher(args, options={})
+      self.class.run_hooks_for(:before_publish, *args)
+      @chore_publisher ||= self.class.options[:publisher]
+      @chore_publisher.publish(self.class.prefixed_queue_name, self.class.job_hash(args), options)
```

Comment:
This will break any existing chore publisher that uses the existing interface which is not in the chore-core gem (so, everything in chore-messaging). You'll want to update those as well.